### PR TITLE
fsrepo cleanup: no more components

### PR DESCRIPTION
**Pull request by tv42**
This simplifies fsrepo to prepare for future changes. The should be no change in behavior.

### CODE REVIEW

File: repo/fsrepo/fsrepo.go [hunk 1]
```diff
@@ -183,11 +316,18 @@ func (r *FSRepo) Close() error {
...
-		}
+	if err := r.closeDatastore(); err != nil {
+		return err
 	}
+
+	// This code existed in the previous versions, but
+	// EventlogComponent.Close was never called. Preserving here
+	// pending further discussion.
+	//
+	// TODO It isn't part of the current contract, but callers may like for us
```
Comment (whyrusleeping) [inline repo/fsrepo/fsrepo.go]: good point

File: repo/fsrepo/fsrepo.go [hunk 2]
```diff
@@ -240,14 +416,39 @@ func (r *FSRepo) SetConfigKey(key string, value interface{}) error {
...
+	if err := common.MapSetKV(mapconf, key, value); err != nil {
+		return err
+	}
+	conf, err := config.FromMap(mapconf)
+	if err != nil {
+		return err
+	}
+	if err := serialize.WriteConfigFile(filename, mapconf); err != nil {
+		return err
+	}
```
Comment (jbenet) [inline repo/fsrepo/fsrepo.go]: btw, if you were wondering why the weird way to write out the config-- the complicating piece here is the user (or other programs) may have added keys to the config that we -- in our types -- dont know about and must not nuke. (our config works like git's).

actually, may be worth adding a comment noting that

**Comment by jbenet** [comment]
This LGTM. @tv42 just gave you contributor access so that you can push branches onto the main repo. our jenkins doesn't pick them up otherwise, so not all tests ran here.

**Comment by jbenet** [comment]
@tv42 btw if you have a better way of doing what he pkg-level lock/counters are doing, go for it. i don't remember the intricacy there, but things will get simpler when the cli does not need to touch the config at all. (this will be fixed when repo implements https://github.com/ipfs/specs/tree/master/repo/fs-repo#api )

**Comment by tv42** [comment]
Closing in favor of https://github.com/jbenet/go-ipfs/pull/917

**Comment by tv42** [comment]
@jbenet Yeah I was eyeing those, but decided to go for a more mechanical transformation of the source for now, to not break anything.

**Pull request closed without merging**